tests/test_updater_top_level_update.py

```diff
@@ -306,6 +308,69 @@ def test_new_timestamp_unsigned(self) -> None:
 
         self._assert_files_exist([Root.type])
 
+    @patch.object(datetime, "datetime", wraps=datetime.datetime)
+    def test_expired_timestamp_version_rollback(self, mock_time: Mock) -> None:
+
+        now = datetime.datetime.utcnow()
+        self.sim.timestamp.expires = now.replace(
+            microsecond=0
+        ) + datetime.timedelta(days=7)
+
+        self.sim.timestamp.version = 2
+
+        # Make a successful update of valid metadata which stores it in cache
+        self._run_refresh()
+
+        self.sim.timestamp.version = 1
+
+        self.sim.timestamp.expires = now.replace(
+            microsecond=0
+        ) + datetime.timedelta(days=21)
+
+        mock_time.utcnow.return_value = (
+            datetime.datetime.utcnow() + datetime.timedelta(days=18)
```

Comment:
```suggestion
            now + datetime.timedelta(days=18)
```
maybe easier to make the connection to the expiry dates if we use the same construct.

This applies to the other tests as well